### PR TITLE
validation: Add for protocol names

### PR DESCRIPTION
**Pull request by evaporei**

### CODE REVIEW

File: src/validation/manifest.js [hunk 1]
```diff
@@ -241,23 +241,61 @@ const validateValue = (value, ctx) => {
...
+// }
+const dataSourceListToMap = dataSources =>
   dataSources
-    .filter(dataSource => protocol.isValidKindName(dataSource.kind))
     .reduce(
-      (networks, dataSource) =>
-        networks.update(dataSource.network, dataSources =>
-          (dataSources || immutable.OrderedSet()).add(dataSource.name),
+      (protocolKinds, dataSource) =>
+        protocolKinds.update(dataSource.kind, networks =>
```
Comment (dimitrovmaksim) [inline src/validation/manifest.js:266]: Currently if you have two datasources, one with kind ethereum and one with ethereum/contract the validation will fail. Unless this is a desired behaviour, a solution could be to replace dataSource.kind with Protocol.fromDataSources([dataSource]).name

File: src/validation/manifest.js [hunk 2]
```diff
@@ -241,23 +241,61 @@ const validateValue = (value, ctx) => {
...
+          (networks || immutable.OrderedMap()).update(dataSource.network, dataSourceNames =>
+            (dataSourceNames || immutable.OrderedSet()).add(dataSource.name)),
         ),
       immutable.OrderedMap(),
     )
 
-const validateDataSourceNetworks = (value, protocol) => {
+const validateDataSourceProtocolAndNetworks = (value, protocol) => {
```
Comment (dimitrovmaksim) [inline src/validation/manifest.js:273]: Do you need the `protocol` argument here, it seems unused?

Comment (evaporei) [reply src/validation/manifest.js]: You're correct! Just removed it 🔫

File: src/validation/manifest.js (hunk 1 above)
Comment (evaporei) [reply src/validation/manifest.js:266]: Woah, good point! I guess I'll need the `protocol` parameter now 😅 

I'll probably use [Protocol.availableProtocols()](https://github.com/graphprotocol/graph-cli/blob/main/src/protocols/index.js#L25)

**Review by dimitrovmaksim — APPROVED** [review]

**Pull request merged**